src/stores/right-panel/RightPanelStore.ts

```diff
@@ -373,45 +368,46 @@ export default class RightPanelStore extends ReadyWatchingStore {
+        // load values from byRoomCache with the viewedRoomId.
+        this.loadCacheFromSettings();
+        this.emitAndUpdateSettings();
     };
 
     onDispatch = (payload: ActionPayload) => {
         switch (payload.action) {
-            case 'view_group':
-            case Action.ViewRoom: {
+            case 'view_group': {
                 if (payload.room_id === this.viewedRoomId) break; // skip this transition, probably a permalink
 
                 // Put group in the same/similar view to what was open from the previously viewed room
                 // Is contradictory to the new "per room" philosophy but it is the legacy behavior for groups.
-                if ((this.isViewingRoom ? Action.ViewRoom : "view_group") != payload.action) {
-                    if (payload.action == Action.ViewRoom && MEMBER_INFO_PHASES.includes(this.currentCard?.phase)) {
-                        // switch from group to room
-                        this.setRightPanelCache({ phase: RightPanelPhases.RoomMemberList, state: {} });
-                    } else if (
-                        payload.action == "view_group" &&
-                        this.currentCard?.phase === RightPanelPhases.GroupMemberInfo
-                    ) {
-                        // switch from room to group
-                        this.setRightPanelCache({ phase: RightPanelPhases.GroupMemberList, state: {} });
-                    }
+
+                if (
+                    this.currentCard?.phase === RightPanelPhases.GroupMemberInfo
+                ) {
+                    // switch from room to group
+                    this.setRightPanelCache({ phase: RightPanelPhases.GroupMemberList, state: {} });
                 }
 
                 // Update the current room here, so that all the other functions dont need to be room dependant.
                 // The right panel store always will return the state for the current room.
                 this.viewedRoomId = payload.room_id;
-                this.isViewingRoom = payload.action == Action.ViewRoom;
+                this.isViewingRoom = false;
                 // load values from byRoomCache with the viewedRoomId.
                 if (this.isReady) {
                     // we need the client to be ready to get the events form the ids of the settings
                     // the loading will be done in the onReady function (to catch up with the changes done here before it was ready)
                     // all the logic in this case is not necessary anymore as soon as groups are dropped and we use: onRoomViewStoreUpdate
                     this.loadCacheFromSettings();
-                    this.emitAndUpdateSettings();
+
+                    /*
+                    DO NOT EMIT. Emitting breaks iframe refs by triggering a render
+                    for the room view and calling the iframe ref changed
+                    function
+                    */
```

Comment:
Maybe line comments instead?

```suggestion
                    // DO NOT EMIT. Emitting breaks iframe refs by triggering a render
                    // for the room view and calling the iframe ref changed function
```